sdk/src/signer/mod.rs

```diff
@@ -103,6 +109,25 @@ pub fn unique_signers(signers: Vec<&dyn Signer>) -> Vec<&dyn Signer> {
     signers.into_iter().unique_by(|s| s.pubkey()).collect()
 }
 
+/// The `EncodableKey` trait defines the interface by which cryptographic keys/keypairs are read,
+/// written, and derived from sources.
+pub trait EncodableKey: Sized {
+    fn pubkey_string(&self) -> Result<String, Box<dyn error::Error>>;
+    fn read_key<R: Read>(reader: &mut R) -> Result<Self, Box<dyn error::Error>>;
+    fn read_key_file<F: AsRef<Path>>(path: F) -> Result<Self, Box<dyn error::Error>>;
+    fn write_key<W: Write>(&self, writer: &mut W) -> Result<String, Box<dyn error::Error>>;
+    fn write_key_file<F: AsRef<Path>>(&self, outfile: F) -> Result<String, Box<dyn error::Error>>;
```

Comment:
`{read,write}_key_file` can be default implementations here on the trait declaration. all they do is open a file then call `{read,write}_key` respectively

Reply:
Agreed!